test/e2e-tests.js

```diff
@@ -379,7 +398,39 @@ describe('matrix-public-archive', () => {
       );
     });
 
-    it(`can render day back in time from room on remote homeserver we haven't backfilled from`);
+    it(`can render day back in time from room on remote homeserver we haven't backfilled from`, async () => {
+      //const hs1Client = await getTestClientForHs(testMatrixServerUrl1);
+      const hs2Client = await getTestClientForHs(testMatrixServerUrl2);
+
+      // Create a room on hs2
+      const hs2RoomId = await createTestRoom(hs2Client);
+      const room2EventIds = await createMessagesInRoom({
+        client: hs2Client,
+        roomId: hs2RoomId,
+        numMessages: 3,
+        prefix: HOMESERVER_URL_TO_PRETTY_NAME_MAP[hs2Client.homeserverUrl],
+        timestamp: archiveDate.getTime(),
+      });
+
+      archiveUrl = matrixPublicArchiveURLCreator.archiveUrlForDate(hs2RoomId, archiveDate, {
+        // Since hs1 doesn't know about this room on hs2 yet, we have to provide
+        // a via server to ask through.
+        viaServers: ['hs2'],
+      });
+
+      const archivePageHtml = await fetchEndpointAsText(archiveUrl);
```

Comment:
This doesn't work yet, thanks to https://github.com/matrix-org/synapse/issues/4778 which means the application service user can't join a remote federated room because it will throw a `404` `Profile was not found` unless it's explicitly registered before. 

And we can't register the username of the application service defined in the registration files. So we would have to do some random dance of removing and adding the application service registration and restarting.